src/components/Menu/MenuItem.tsx

```diff
@@ -60,7 +60,7 @@ type Props<C extends MenuItemElement> = {
    * Event callback for when the item is clicked.
    * @param e
    */
-  onClick?: MouseEventHandler<C>;
+  onClick?: MouseEventHandler;
```

Comment:
```suggestion
  onClick?: MouseEventHandler<HTMLElementTagNameMap[C]>;
```

maybe? (untested)

Reply:
It's working :)

TIL `HTMLElementTagNameMap`

Reply:
We use it in AccessibleButton in element-web FTR